### PR TITLE
Use pom type for flow dependency

### PR DESCRIPTION
**Pull request by denis-anisimov**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/spring/319)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -61,6 +61,7 @@
                 <groupId>com.vaadin</groupId>
                 <artifactId>flow</artifactId>
                 <version>${vaadin.flow.version}</version>
+                <type>pom</type>
```
Comment (snicoll) [inline pom.xml]: If the intend is to import `flow` as a bom, then it's missing the scope import. Importing a pom as a regular dependency is very unusual. Can you please share a bit more context about this?

Comment (snicoll) [reply pom.xml]: I see, thanks for the feedback. Agreed that importing a `pom` artifact like that is a smell and should be fixed in its own right.

**Comment by denis-anisimov** [comment]
Review status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @tsuoanttila)

---

*[pom.xml, line 64 at r1](https://reviewable.io/reviews/vaadin/spring/319#-LFH_yrFk3QH4lvYb8bB-r1-64:-LFHe-Fe09H-hj7kqCxK:b563yx8) ([raw file](https://github.com/vaadin/spring/blob/abc647d20107c39d9311645b5835cbf2d7315354/pom.xml#L64)):*
<details><summary><i>Previously, snicoll (Stéphane Nicoll) wrote…</i></summary><blockquote>

I see, thanks for the feedback. Agreed that importing a `pom` artifact like that is a smell and should be fixed in its own right.
</blockquote></details>

So what's the resulting resolution?
Do we keep it as is for a while (as it's already done for some components).
Or should I do to fix it?
Spring cannot be released at the moment since the build is failing.
Those are blocking comments which prevents this PR to be merged.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/spring/319)*

**Comment by ZheSun88** [comment]
Review status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @snicoll)

---

*[pom.xml, line 64 at r1](https://reviewable.io/reviews/vaadin/spring/319#-LFH_yrFk3QH4lvYb8bB-r1-64:-LFHfzUU7X0o4hgcSmGC:b674q8a) ([raw file](https://github.com/vaadin/spring/blob/abc647d20107c39d9311645b5835cbf2d7315354/pom.xml#L64)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

So what's the resulting resolution?
Do we keep it as is for a while (as it's already done for some components).
Or should I do to fix it?
Spring cannot be released at the moment since the build is failing.
Those are blocking comments which prevents this PR to be merged.
</blockquote></details>

I think the proper fix for this is replace the flow dependency to the actual flow-server, flow-html-component, etc.. which are the ones used in the project

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/spring/319)*

**Comment by denis-anisimov** [comment]
Review status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @snicoll)

---

*[pom.xml, line 64 at r1](https://reviewable.io/reviews/vaadin/spring/319#-LFH_yrFk3QH4lvYb8bB-r1-64:-LFHgvAJ8cdt7lkz4y7v:b-ycsh4v) ([raw file](https://github.com/vaadin/spring/blob/abc647d20107c39d9311645b5835cbf2d7315354/pom.xml#L64)):*
<details><summary><i>Previously, ZheSun88 (Sun Zhe) wrote…</i></summary><blockquote>

I think the proper fix for this is replace the flow dependency to the actual flow-server, flow-html-component, etc.. which are the ones used in the project
</blockquote></details>

OK......
Why do we have this then: 
https://github.com/vaadin/vaadin-checkbox-flow/blob/master/pom.xml#L40

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/spring/319)*

**Comment by ZheSun88** [comment]
Review status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @snicoll)

---

*[pom.xml, line 64 at r1](https://reviewable.io/reviews/vaadin/spring/319#-LFH_yrFk3QH4lvYb8bB-r1-64:-LFHhRAHAjynzU57W3Up:b-w63n9i) ([raw file](https://github.com/vaadin/spring/blob/abc647d20107c39d9311645b5835cbf2d7315354/pom.xml#L64)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

OK......
Why do we have this then: 
https://github.com/vaadin/vaadin-checkbox-flow/blob/master/pom.xml#L40
</blockquote></details>

this should be fixed also... as Teemu has already create the ticket to remove the flow module (https://github.com/vaadin/flow/issues/4287), that is just a temporary workaround

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/spring/319)*

**Comment by denis-anisimov** [comment]
Review status: 0 of 7 files reviewed, 1 unresolved discussion (waiting on @snicoll)

---

*[pom.xml, line 64 at r1](https://reviewable.io/reviews/vaadin/spring/319#-LFH_yrFk3QH4lvYb8bB-r1-64:-LFHk5Xa2S_zBmrHdXbL:bb6un1j) ([raw file](https://github.com/vaadin/spring/blob/abc647d20107c39d9311645b5835cbf2d7315354/pom.xml#L64)):*
<details><summary><i>Previously, ZheSun88 (Sun Zhe) wrote…</i></summary><blockquote>

this should be fixed also... as Teemu has already create the ticket to remove the flow module (https://github.com/vaadin/flow/issues/4287), that is just a temporary workaround
</blockquote></details>

Done

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/spring/319)*

**Comment by tsuoanttila** [comment]
Review status: 0 of 7 files reviewed, 1 unresolved discussion (waiting on @snicoll)

---

*[pom.xml, line 64 at r1](https://reviewable.io/reviews/vaadin/spring/319#-LFH_yrFk3QH4lvYb8bB-r1-64:-LFHl8u-3jLdLsVoQJhe:b-wyoiv2) ([raw file](https://github.com/vaadin/spring/blob/abc647d20107c39d9311645b5835cbf2d7315354/pom.xml#L64)):*
<details><summary><i>Previously, denis-anisimov (Denis) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Looks fine by me.

---


*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/spring/319)*

**Comment by ZheSun88** [comment]
:lgtm:

---

Reviewed 7 of 7 files at r2.
Review status: all files reviewed, 1 unresolved discussion (waiting on @snicoll)

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/spring/319#-:-LFHya010Zu59fxfKhuD:bnfp4nl)*

**Comment by denis-anisimov** [comment]
Dismissed @snicoll from a discussion.
Review status: :shipit: complete! all files reviewed, all discussions resolved

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/spring/319)*

**Pull request merged**